src/WebJobs.Extensions.ApiHub/Listener/ApiHubListener.cs

```diff
@@ -57,13 +57,11 @@ public Task StartAsync(CancellationToken cancellationToken)
 
             if (_poll == null)
             {
-                var errorText = string.Format("Path '{0}' is invalid. IFolderItem.RootPath must be set to a valid directory location.", _folderSource.Path);
-                _trace.Error(errorText);
-
-                throw new InvalidOperationException(errorText);
+                var errorText = string.Format("Path '{0}' is invalid. Path must be set to a valid directory location.", _folderSource.Path);
+                var ex = new InvalidOperationException(errorText);
+                _trace.Error(errorText, ex);
```

Comment:
Ok making the changes in the core SDK to allow listener exceptions to be "handled" is very reasonable. Please let me know if you want me to close this PR then.